components/fxa-client/ffi/src/lib.rs

```diff
@@ -224,6 +224,126 @@ pub unsafe extern "C" fn fxa_get_access_token(
     })
 }
 
+/// Update the Push subscription information for the current device.
+#[no_mangle]
+pub unsafe extern "C" fn fxa_set_push_subscription(
+    handle: u64,
+    endpoint: *const c_char,
+    public_key: *const c_char,
+    auth_key: *const c_char,
+    error: &mut ExternError,
+) {
+    log::debug!("fxa_set_push_subscription");
+    ACCOUNTS.call_with_result_mut(error, handle, |fxa| {
+        let ps = PushSubscription {
+            endpoint: rust_str_from_c(endpoint).to_owned(),
+            public_key: rust_str_from_c(public_key).to_owned(),
+            auth_key: rust_str_from_c(auth_key).to_owned(),
+        };
+        // We don't really care about passing back the resulting Device record.
+        // We might in the future though.
+        fxa.set_push_subscription(&ps).map(|_| ())
+    })
+}
+
+/// Update the display name for the current device.
+#[no_mangle]
+pub unsafe extern "C" fn fxa_set_display_name(
+    handle: u64,
+    display_name: *const c_char,
+    error: &mut ExternError,
+) {
+    log::debug!("fxa_set_display_name");
+    ACCOUNTS.call_with_result_mut(error, handle, |fxa| {
+        // We don't really care about passing back the resulting Device record.
+        // We might in the future though.
+        fxa.set_display_name(rust_str_from_c(display_name))
+            .map(|_| ())
+    })
+}
+
+/// Fetch the devices (including the current one) in the current account.
+///
+/// # Safety
+///
+/// A destructor [fxa_bytebuffer_free] is provided for releasing the memory for this
+/// pointer type.
+#[no_mangle]
+pub unsafe extern "C" fn fxa_get_devices(handle: u64, error: &mut ExternError) -> ByteBuffer {
+    log::debug!("fxa_get_devices");
+    ACCOUNTS.call_with_result_mut(error, handle, |fxa| {
+        fxa.get_devices().map(|d| {
+            let devices = d.into_iter().map(|device| device.into()).collect();
+            fxa_client::msg_types::Devices { devices }
+        })
+    })
+}
+
+/// Poll and handle available remote commands targeted to our own device.
```

Comment:
nit: do we really "handle" them, or should this docstring say something like "return events to be handled by the calling code"?